portlist/netstat.go

```diff
@@ -41,50 +43,57 @@ func parsePort(s string) int {
 	return int(port)
 }
 
-func isLoopbackAddr(s string) bool {
-	return strings.HasPrefix(s, "127.") ||
-		strings.HasPrefix(s, "[::1]:") ||
-		strings.HasPrefix(s, "::1.")
+func isLoopbackAddr(s mem.RO) bool {
+	return mem.HasPrefix(s, mem.S("127.")) ||
+		mem.HasPrefix(s, mem.S("[::1]:")) ||
+		mem.HasPrefix(s, mem.S("::1."))
 }
 
 type nothing struct{}
 
-// Lowest common denominator parser for "netstat -na" format.
+// This used to be a lowest common denominator parser for "netstat -na" format.
 // All of Linux, Windows, and macOS support -na and give similar-ish output
 // formats that we can parse without special detection logic.
 // Unfortunately, options to filter by proto or state are non-portable,
 // so we'll filter for ourselves.
-func appendParsePortsNetstat(base []Port, output string) []Port {
-	m := map[Port]nothing{}
-	lines := strings.Split(string(output), "\n")
-
-	var lastline string
-	var lastport Port
-	for _, line := range lines {
-		trimline := strings.TrimSpace(line)
-		cols := strings.Fields(trimline)
+// Nowadays, though, we only use it for macOS as of 2022-11-04.
+func appendParsePortsNetstat(base []Port, br *bufio.Reader) ([]Port, error) {
+	ret := base
```

Comment:
Add some example lines in a comment, to make walking the parser logic easier?

Reply:
Oh, or a comment pointing to tests for example inputs would do.

Reply:
Done.